piq/utils.py

```diff
@@ -18,7 +18,7 @@ def _adjust_dimensions(input_tensors: Union[torch.Tensor, Tuple[torch.Tensor, to
             tmp = tmp.unsqueeze(0)
         if tmp.dim() != 4 and tmp.dim() != 5:
             raise ValueError(f'Expected 2, 3, 4 or 5 dimensions (got {tensor.dim()})')
-        resized_tensors.append(tmp)
+        resized_tensors.append(tmp.type(torch.float32))
```

Comment:
Why we need to change tensor type during resize? 
This should be mentioned in docstring

Reply:
Yeah, I noticed this one as well. 
We have not addressed data types of passed tensors and to me it seems like too much of work to do that now (because then all data types will need to be tested etc.)

I agree with @zakajd that if you change data type it need to be explicitly defined several times in different places or (even better) just be avoided. What is the use case when the absence of this conversion break things down?

Reply:
This is the case for SSIM and MS-SSIM. We usually use `.to(prediction)`, but it doesn't work here in case of prediction with `dtype=torch.uint8`. For example, it is impossible to create gaussian filter with `dtype=torch.uint8`. 
Next, we will have:
- the rounded result is equal `1` instead of `0.6824` in case of SSIM;
- conflict at `torch.nn.functional.pad` in case of MS-SSIM.

Almost all operations return `float` results. The translation to `torch.float32` is already used as a result of `x = x / float(data_range)`.
Since we implicitly change the type of the input tensors, I propose to make it at `_validation_input` explicitly and show the corresponding warning.

Reply:
Agh I would still avoid doing it.   
If one has `torch.ByteTensor` prediction, let him/her convert it to `torch.float32` and then compute SSIM/MS-SSIM with it. This whole type conversion thing is out of the scope of this lib.

Reply:
@snk4tr, `torch.ByteTensor` is `torch.uint8`, which is usual data type for images. Let's take `FSIM`. We can pass `torch.uint8` tensor and got the answer. This is the case for all metrics meaning that we already provide user with such implicit conversion. What is more, it is not logical making one metrics unavailable because of wrong type and keeping other metrics work with it. I move the conversion especially for torch.uint8 inside SSIM.

Reply:
@snk4tr @zakajd What do you think about strict assert of the input data type? One can pass the `ByteTensor` to a measure, but they cannot use them for back propagation (gradients are available only for `FloatTensor`).